### DISCUSSION TITLE
Private not declared by package fx

### DISCUSSION THREAD

**Discussion by pfernandom** [uber-go/fx#1007] [Q&A] [answered]
Hello,

I'm trying to build a module, keeping its intermediate dependencies private:

```
import (
	"go.uber.org/fx"
)

// ...

fx.Module("mymodule",
        //...
	fx.Provide(
		fx.Private,
		cases.NewPrivateService,
	),
)
```

However, the compiler is complaining:

```
Private not declared by package fx  compiler[UndeclaredImportedName](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UndeclaredImportedName)
```

I'm using fx version:

```
go.uber.org/fx v1.18.2
```

What am I doing wrong?

**Comment by pfernandom** [comment]
Ahh nvm, seems like `Private` is not exported in the latest version. I had to take the code from master

**Comment by sywhang** [comment, marked as answer]
Yes, `Private` was made available since v1.19.0. Please use the latest release (v1.19.1) for you to make use of this feature.
